fire_atlas_editor/lib/store/actions/atlas_actions.dart

```diff
@@ -40,42 +40,47 @@ class CreateAtlasAction extends AsyncMicroStoreAction<FireAtlasState> {
 class UpdateAtlasImageAction extends AsyncMicroStoreAction<FireAtlasState> {
   final String imageData;
 
-  UpdateAtlasImageAction({this.imageData});
+  UpdateAtlasImageAction({required this.imageData});
 
   @override
   Future<FireAtlasState> perform(state) async {
-    state
-      ..currentAtlas.imageData = imageData
-      ..hasChanges = true;
+    if (state.currentAtlas != null) {
+      final atlas = state.currentAtlas!;
 
-    await state.currentAtlas.loadImage(clearImageData: false);
+      state.hasChanges = true;
+      atlas.imageData = imageData;
+      await atlas.loadImage(clearImageData: false);
+    }
 
     return state;
   }
 }
 
 class SetSelectionAction extends MicroStoreAction<FireAtlasState> {
-  Selection selection;
+  BaseSelection selection;
 
   SetSelectionAction({
-    @required this.selection,
+    required this.selection,
   });
 
   @override
   FireAtlasState perform(FireAtlasState state) {
-    state.currentAtlas.selections[selection.id] = selection;
-    state.selectedSelection = selection;
-    state.hasChanges = true;
+    if (state.currentAtlas != null) {
+      final atlas = state.currentAtlas!;
```

Comment:
ordering, avoid !

Reply:
Early in, early out :-)

Reply:
> ordering, avoid!

Fixed

> Early in, early out :-)

I always return something in the end, so for this case I still prefer to use if since it is just one condition.